js/indicators/awesome.src.js

```diff
@@ -0,0 +1,197 @@
+     *                     Awesome
+     * @since 7.0.0
+     * @excluding
+     *             allAreas,colorAxis,joinBy,keys,stacking,
+     *             showInNavigator,navigatorOptions,params,pointInterval,
+     *             pointIntervalUnit,pointPlacement,pointRange,pointStart
+     * @optionparent plotOptions.awesome
+     */
+    {
+        /**
+         * Color of the awesome series bar that is greater than the
+         * previous one. Note that if a `color` is defined, the `color`
+         * takes precedence and the `greaterBarColor` is ignored.
+         *
+         * @sample {highstock} stock/indicators/awesome/
+         *          greaterBarColor
+         *
+         * @type {Highcharts.ColorString}
+         * @since 7.0.0
+         */
+        greaterBarColor: '#06B535',
+        /**
+         * Color of the awesome series bar that is lower than the
+         * previous one. Note that if a `color` is defined, the `color`
+         * takes precedence and the `lowerBarColor` is ignored.
+         *
+         * @sample {highstock} stock/indicators/awesome/
+         *          lowerBarColor
+         *
+         * @type {Highcharts.ColorString}
+         * @since 7.0.0
+         */
+        lowerBarColor: '#F21313',
+        threshold: 0,
+        groupPadding: 0.2,
+        pointPadding: 0.2,
+        states: {
+            hover: {
+                halo: {
+                    size: 0
+                }
+            }
+        }
+    }, /** @lends Highcharts.Series.prototype */ {
+        nameBase: 'AO',
+        nameComponents: false,
+
+        // Columns support:
+        markerAttribs: noop,
+        getColumnMetrics: H.seriesTypes.column.prototype.getColumnMetrics,
+        crispCol: H.seriesTypes.column.prototype.crispCol,
+        translate: H.seriesTypes.column.prototype.translate,
+        drawPoints: H.seriesTypes.column.prototype.drawPoints,
+
+        drawGraph: function () {
+            var indicator = this,
+                options = indicator.options,
+                points = indicator.points,
+                userColor = indicator.userOptions.color,
+                positiveColor = options.greaterBarColor,
+                negativeColor = options.lowerBarColor,
+                firstPoint = points[0],
+                i;
+
+            if (!userColor && firstPoint) {
+                firstPoint.color = positiveColor;
+
+                for (i = 1; i < points.length; i++) {
+                    if (points[i].y > points[i - 1].y) {
+                        points[i].color = positiveColor;
+                    } else if (points[i].y < points[i - 1].y) {
+                        points[i].color = negativeColor;
+                    } else {
+                        points[i].color = points[i - 1].color;
+                    }
+                }
+            }
+        },
+
+        getValues: function (series) {
+            var shortPeriod = 5,
+                longPeriod = 34,
```

Comment:
Why is it hardcoded?

Reply:
It is hardcoded because awesome oscillator has fixed periods of 5 and 34.